lib/ecto/repo/queryable.ex

```diff
@@ -14,6 +14,8 @@ defmodule Ecto.Repo.Queryable do
     adapter.transaction(repo, opts, fun)
   end
 
+  def transaction(_adapter, _repo, %Ecto.Multi{operations:  []}, _opts),
+    do: {:ok, %{}}
```

Comment:
I would prefer we do this in the `Ecto.Multi` module instead of here.

The `Multi` struct is defined as opaque, so we shouldn't destructure it outside the defining module.

Appropriate place would be here: https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/multi.ex#L378